src/rules/no-duplicate-definitions.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * @fileoverview Rule to prevent duplicate definitions in Markdown.
+ * @author 루밀LuMir(lumirlumir)
+ */
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/** @typedef {import("mdast").Definition} Definition */
+/** @typedef {import("mdast").FootnoteDefinition} FootnoteDefinition */
+/**
+ * @typedef {import("../types.ts").MarkdownRuleDefinition<{ RuleOptions: [{ ignoreDefinition: string[], ignoreFootnoteDefinition: string[]; }]; }>}
+ * NoDuplicateDefinitionsRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+/**
+ * Appends a node to a map, grouping by identifier.
+ * Creates a new array if the identifier doesn't exist,
+ * or appends to the existing array if it does.
+ * @param {Map<string, Array<Definition | FootnoteDefinition>>} map The map to store nodes in.
+ * @param {Definition | FootnoteDefinition} node The node to add to the map.
+ * @returns {void}
+ */
+function appendNodeToMap(map, node) {
+	map.set(
+		node.identifier,
+		map.has(node.identifier) ? [...map.get(node.identifier), node] : [node],
+	);
```

Comment:
This creates a new array every time. Instead, we mutate the existing array:
```suggestion
	if (!map.has(node.identifier)) {
      map.set(node.identifier, [node]);
    } else {
      map.get(node.identifier).push(node);
    }
```